draft-ietf-lamps-rfc4210bis.md

```diff
@@ -1170,131 +1182,91 @@ other things) the certificate containing the public key of the
-            in order     using the     the            verify the
-            to get the   repository    repository;    certificate
-            value of                   however, the   directly -
-            the OLD                    verification   this is thus
-            public key                 will FAIL      the same as
-                                                      case 4.
-~~~~
+|     | Verifier's PSE contains NEW public key | Verifier's PSE contains OLD public key |
+|:----|:---------------------------------------|:---------------------------------------|
+| Signer's certificate is protected using NEW key pair | Case 1: The verifier can directly verify the certificate. | Case 2: The verifier is missing the NEW public key. |
+| Signer's certificate is protected using OLD key pair | Case 3: The verifier is missing the OLD public key. | Case 4: The verifier can directly verify the certificate. |
 
-Note: Instead of using a repository, the end entity can use the root CA update
-general message to request the respective certificates from a PKI management
-entity, see {{sect-5.3.19.15}}, and follow the required validation steps.
 
-#### Verification in Cases 1, 4, 5, and 8
+#### Verification in Cases 1 and 4
 {: id="sect-4.4.2.1"}
 
 In these cases, the verifier has a local copy of the CA public key
 that can be used to verify the certificate directly.  This is the
 same as the situation where no key change has occurred.
 
-Note that case 8 may arise between the time when the CA operator has
-generated the new key pair and the time when the CA operator stores
-the updated attributes in the repository.  Case 5 can only arise if
-the CA operator has issued both the signer's and verifier's
-certificates during this "gap" (the CA operator SHOULD avoid this as
-it leads to the failure cases described below)
-
 
 #### Verification in Case 2
 {: id="sect-4.4.2.2"}
 
-In case 2, the verifier must get access to the old public key of the
-CA.  The verifier does the following:
+In case 2, the verifier must get access to the new public key of the
+CA. Case 2 will arise when the CA operator has issued the verifier's
+certificate, then changed the CA's key, and then issued the signer's
+certificate; so it is quite a typical case.
 
+The verifier does the following:
 
-1. Look up the caCertificate attribute in the repository and pick
-  the OldWithNew certificate (determined based on validity periods;
-  note that the subject and issuer fields must match);
 
-1. Verify that this is correct using the new CA key (which the
-  verifier has locally);
+1. Get the "new with new" and "new with old" certificate.
 
-1. If correct, check the signer's certificate using the old CA key.
 
-Case 2 will arise when the CA operator has issued the signer's
-certificate, then changed the key, and then issued the verifier's
-certificate; so it is quite a typical case.
 
+    1. If a repository is available, look up the caCertificate
```

Comment:
Couldn't they get the bundle of certificates (new with new, new with old, etc using the AIA v3 extension?)  if it exists in the existing certificate.  Usually it points to an http .p7c file, which could be updated by the CA when the "new with new" and "new with old" is created.

Reply:
I was not aware of this option. We could add it as a third option, if it really works.

Reply:
Is there an RFC specifying this usage of the AIA, or is it pure RFC 5280?

Reply:
Yes, I think it can be another option.   The purpose is to enable the client a way to lookup the certificates needed for path building (in this case the new CA cert and the CA link certificate), and RFC 5280 section 4.2.2.1 describes how id-ad-caIssuers works.   So I don't think there is a reason why it couldn't be used, its just a more standard way of looking up certificates (rather than other out-of-band mechanisms that are out of scope for this document).

Reply:
I don't know of any specific RFC that specifies AIA usage for CA key renewal, I think RFC 4210 was based on RFC 2510, which was around the time of RFC 3280.  I see AIA ca-Issuers was in there, but maybe it was too new at the time, or they just didn't think about using it?

Reply:
@johngray-dev  What do you fink of this wording?

~~~
The verifier does the following:
1. Get the "new with new" and "new with old" certificate.
    1. If an caIssuers access method is available in the authority
      information access extension of the "old with old" certificate,
      the verifier should retrieve the "new with new" and
      "new with new" certificate using the specified assess location,
      see Section 4.2.2.1 of {{RFC5280}}.
    2. If no caIssuers access method is available, the verifier can
      use the root CA update general message to request the new root
      CA and link certificates from a PKI management entity, see
      {{sect-5.3.19.15}}.
    3. Otherwise, get it "out-of-band" in a trustworthy manner.
2. If received the two certificates, check that the validity periods
  and the subject and issuer fields match. Verify the signatures
  using the old CA key (which the verifier has locally). Securely
  store the new trust anchor information if all checks were
  successful.
3. If correct, check the signer's certificate using the new CA key.
~~~

Reply:
alternative proposal
~~~
The verifier does the following:
1. Get the "new with new" and "new with old" certificate.
    1. The verifier may use an URI to retrieve the "new with new" and
      "new with new" certificates as described in Section 4.2.2.1 of
      {{RFC5280}}. Such URIs may be provided in the caIssuers method
      of the authority information access extension of the
      "old with old" certificate, or may be configured locally.
    2. The verifier may use PKIX-CMP the root CA update general message to request the new root
      CA and link certificates from a PKI management entity, see
      {{sect-5.3.19.15}}.
    3. Otherwise, the verifier may get it "out-of-band" in any trustworthy manner.
2. If received the two certificates, check that the validity periods
  and the subject and issuer fields match. Verify the signatures
  using the old CA key (which the verifier has locally). Securely
  store the new trust anchor information if all checks were
  successful.
3. If correct, check the signer's certificate using the new CA key.
~~~

Reply:
I updated the PR